content/blog/purrr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,485 @@
+Then we'll get to the good stuff: improvements to the `map` family, new `keep_at()` and `discard_at()` functions, and improvements to flattening and simplification.
+You can see a full list of changes in the [release notes](%7B%20github_release%20%7D).
+
+```{r setup}
+library(purrr)
+```
+
+## Breaking changes
+
+We've used the 1.0.0 release as an opportunity to really refine the core purpose of purrr: facilitating functional programming in R.
+We've been more aggressive with deprecations and breaking changes than usual, because a 1.0.0 release signals that purrr is now [stable](https://lifecycle.r-lib.org/articles/stages.html#stable), making it our last opportunity for major changes.
+
+These changes will break some existing code, but we've done our best to make it affect as little code as possible.
+Out of the \~1400 CRAN packages that user purrr, only \~40 were negatively affected, and I [made pull requests](https://github.com/tidyverse/purrr/issues/969) to fix them all.
+Making these fixes helped give me confidence that, though we're deprecating quite a few functions and changing a few special cases, it shouldn't affect too much code in the wild.
+
+There are four important changes that you should be aware of:
+
+-   `pluck()` behaves differently when extracting 0-length vectors.
+-   The `map()` family uses the tidyverse rules for coercion and recycling.
+-   All functions that modify lists handle `NULL` consistently.
+-   We've deprecated functions that aren't related to the core purpose of purr.
+
+### pluck and zero-length vectors
+
+Previously, `pluck()` replaced 0-length vectors with the value of `default`.
+Now `default` is only used for `NULL`s and absent elements:
+
+```{r}
+x <- list(y = list(a = character(), b = NULL))
+x |> pluck("y", "a", .default = NA)
+x |> pluck("y", "b", .default = NA)
+x |> pluck("y", "c", .default = NA)
+```
+
+This also influences the map family because using an integer vector, character vector, or list instead of a function automatically calls `pluck()`:
+
+```{r}
+x <- list(list(1), list(), list(NULL), list(character()))
+x |> map(1, .default = 0) |> str()
+```
+
+We made this change because it makes purrr more consistent with the rest of the tidyverse and it looks like it was a bug in the original implementation of the function.
+
+### Tidyverse consistency
+
+We've tweaked the map family of functions to be more consistent with general tidyverse coercion and recycling rules, as implemented by the [vctrs](https://vctrs.r-lib.org) package.
+`map_lgl()`, `map_int()`, `map_int()`, and `map_dbl()` now follow the same [coercion rules](https://vctrs.r-lib.org/articles/type-size.html#coercing-to-common-type) as vctrs.
+In particular:
+
+-   `map_chr(TRUE, identity)`, `map_chr(0L, identity)`, and `map_chr(1L, identity)` have been deprecated because we believe that converting a logical/integer/double to a character vector is potentially dangerous and should require an explicit coercion.
```

Comment:
```suggestion
-   `map_chr(TRUE, identity)`, `map_chr(0L, identity)`, and `map_chr(1, identity)` have been deprecated because we believe that converting a logical/integer/double to a character vector is potentially dangerous and should require an explicit coercion.
```

Did you mean to show a double here? Or just 0L vs 1L?

Reply:
I meant to show a double, maybe I should make it 1.5.